commit 6aed423a0e54306003228d3b68196c542af6cd59
Author: Philipp Zabel <p.zabel@pengutronix.de>
Date:   Mon Jan 4 17:30:09 2016 -0200

    coda: fix first encoded frame payload
    
    commit 74dc385cb450089b28c28be2c8a0baca296b95f9 upstream.
    
    During the recent vb2_buffer restructuring, the calculation of the
    buffer payload reported to userspace was accidentally broken for the
    first encoded frame, counting only the length of the headers.
    This patch re-adds the length of the actual frame data.
    
    Fixes: 2d7007153f0c ("[media] media: videobuf2: Restructure vb2_buffer")
    
    Reported-by: Michael Olbrich <m.olbrich@pengutronix.de>
    Signed-off-by: Philipp Zabel <p.zabel@pengutronix.de>
    Tested-by: Jan Luebbe <jlu@pengutronix.de>
    Signed-off-by: Hans Verkuil <hans.verkuil@cisco.com>
    Signed-off-by: Mauro Carvalho Chehab <mchehab@osg.samsung.com>
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>

diff --git a/drivers/media/platform/coda/coda-bit.c b/drivers/media/platform/coda/coda-bit.c
index 654e964..d76511c 100644
--- a/drivers/media/platform/coda/coda-bit.c
+++ b/drivers/media/platform/coda/coda-bit.c
@@ -1342,7 +1342,7 @@ static void coda_finish_encode(struct coda_ctx *ctx)
 
 	/* Calculate bytesused field */
 	if (dst_buf->sequence == 0) {
-		vb2_set_plane_payload(&dst_buf->vb2_buf, 0,
+		vb2_set_plane_payload(&dst_buf->vb2_buf, 0, wr_ptr - start_ptr +
 					ctx->vpu_header_size[0] +
 					ctx->vpu_header_size[1] +
 					ctx->vpu_header_size[2]);
diff -uN a/1.txt b/1.txt
--- a/dummy/rpi_1339_6aed423a0e54306003228d3b68196c542af6cd59.txt	1970-01-01 00:00:00.000000000 +0000
+++ b/dummy/rpi_1339_6aed423a0e54306003228d3b68196c542af6cd59.txt	2013-12-23 04:07:40.000000000 +0000
@@ -0,0 +1 @@
+dummy file to ensure patch has content.
